x-pack/plugins/security_solution/public/timelines/components/flyout/header/index.tsx

```diff
@@ -227,38 +244,109 @@ const TimelineStatusInfoComponent: React.FC<FlyoutHeaderProps> = ({ timelineId }
 
 const TimelineStatusInfo = React.memo(TimelineStatusInfoComponent);
 
-const FlyoutHeaderComponent: React.FC<FlyoutHeaderProps> = ({ timelineId }) => (
-  <StyledTimelineHeader alignItems="center">
-    <EuiFlexItem>
-      <EuiFlexGroup data-test-subj="properties-left" direction="column" gutterSize="none">
-        <RowFlexItem>
-          <TimelineName timelineId={timelineId} />
-          <SaveTimelineButton timelineId={timelineId} initialFocus="title" />
-        </RowFlexItem>
-        <RowFlexItem>
-          <TimelineDescription timelineId={timelineId} />
-          <SaveTimelineButton timelineId={timelineId} initialFocus="description" />
-        </RowFlexItem>
-        <EuiFlexItem>
-          <TimelineStatusInfo timelineId={timelineId} />
-        </EuiFlexItem>
-      </EuiFlexGroup>
-    </EuiFlexItem>
+const FlyoutHeaderComponent: React.FC<FlyoutHeaderProps> = ({ timelineId }) => {
+  const { selectedPatterns, indexPattern, docValueFields, browserFields } = useSourcererScope(
+    SourcererScopeName.timeline
+  );
+  const getStartSelector = useMemo(() => startSelector(), []);
+  const getEndSelector = useMemo(() => endSelector(), []);
+  const isActive = useMemo(() => timelineId === TimelineId.active, [timelineId]);
+  const from = useDeepEqualSelector((state) => {
```

Comment:
This isn't specific to your PR, but I wonder what the performance improvement we get for the useDeepEqual vs just useSelector. Like are all these deep checks more costly than just re-rendering? Also, maybe just combining all of them into one, so you're only doing one deep check? 🤷🏾‍♂️